### ISSUE TITLE
Default dicts in params is very bad practice

### ISSUE THREAD

**Issue by thehesiod** [horejsek/python-fastjsonschema#192]
see https://github.com/horejsek/python-fastjsonschema/blob/master/fastjsonschema/__init__.py#L149

the empty dict defaults will only be processed once, and are mutable, which means the function now has state.  Should be `None` and then coalesced to empty dict in the function

**Comment by horejsek** [comment]
I agree that None is more correct. At the same time, it is treated as immutable data and not changed by the library. For that reason, I prefer to keep dict as its more clear for auto-generated documentation what type is expected.

**Comment by thehesiod** [comment]
there are better ways to document defaults
